tests/virt_autotest/Prj1_Guest_Installation/guest_installation_run.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+use lib "/var/lib/openqa/share/tests/sle-12-SP2/tests/virt_autotest/lib";
+use base "teststepapi";
+
+use testapi;
+
+our $PRODUCT_TESTED_ON = "SLES-12-SP2";
+our $PROJECT_NAME = "GuestIn_stallation";
+
+
+sub get_guest_pattern() {
+    my $self = shift;
+    my $guest_pattern = get_var("GUEST_PATTERN", "");
+    if ( $guest_pattern eq "") {
+        #$guest_pattern = "sles-11-sp[34]|sles-12";
+        #$guest_pattern = "sles-12-sp2-64-[fp]v-def-net";
+        $guest_pattern = "sles-11-sp3-64-pv-def-net";
+    }
+    return $guest_pattern
+}
```

Comment:
The whole function seems redundant if you just apply a default value to get_var, i.e. replace the whole function where it is called by

```perl
get_var('GUEST_PATTERN', 'sles-11-sp3-64-pv-def-net');
```
but even then the question is why should it be `sles-11-sp3-64-pv-def-net`. Maybe not set the variable default at all in the test and just set it over `vars.json` or correspondingly in the database settings.